includes/db.inc

```diff
@@ -0,0 +1,97 @@
+<?php
+
+/**
+ * @file
+ * Database functionality for tracking job status in Islandora Job.
+ */
+
+/**
+ * Inserts a job to have its state tracked.
+ *
+ * @param string $data
+ *   A JSON encoded string representing the payload data.
+ *
+ * @return int
+ *   The ID for the job's status in the database.
+ */
+function islandora_job_insert_job_status($data) {
+  return db_insert('islandora_job_job_status')
+    ->fields(array(
+      'data' => $data,
+    ))
+    ->execute();
+}
+
+/**
+ * Updates a job's state.
+ *
+ * @param string $job_id
+ *   The ID of the job being tracked to be updated.
+ * @param int $status
+ *   The state of the job being created or updated.
+ */
+function islandora_job_update_job_status($job_id, $status) {
+  db_update('islandora_job_job_status')
+    ->fields(array(
+      'state' => $status,
+    ))
+    ->condition('status_id', $job_id)
+    ->execute();
+}
+
+/**
+ * Updates a job's state given a payload.
+ *
+ * @param string $data
+ *   The data of the payload being used to identify the job.
+ *
+ * @return int
+ *   The ID of the status to be used for removal later.
+ */
+function islandora_job_start_job_status($data) {
+  // Get the status ID of the database to use to remove itself later on.
```

Comment:
Should have an XXX to note is not a race due to usage so doesn't need the transaction overhead (yet).